pkg/reconciler/contour/resources/httpproxy.go

```diff
@@ -97,19 +97,31 @@ func defaultRetryPolicy() *v1.RetryPolicy {
 	}
 }
 
+func tlsEntries(ing *v1alpha1.Ingress) map[string]v1alpha1.IngressTLS {
+	external := ing.GetIngressTLSForVisibility(v1alpha1.IngressVisibilityExternalIP)
+	internal := ing.GetIngressTLSForVisibility(v1alpha1.IngressVisibilityClusterLocal)
+
+	entries := make(map[string]v1alpha1.IngressTLS, len(external)+len(internal))
+	for _, tls := range external {
```

Comment:
Maybe for more readability you could avoid repeating the loop using a local func and then do:

```
addHostEntries(internal)
addHostEntries(external)
```